File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> prefixSums = new HashMap<>();
      
        prefixSums.put(0, 1);
      
        int count = 0;
        int sum = 0;
      
        for (int num : nums) {
            sum += num;
          
            count += prefixSums.getOrDefault(sum - k, 0);
          
            prefixSums.put(sum, prefixSums.getOrDefault(sum, 0) + 1);
        }
      
        return count;

    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cs
public class Solution {
    public int SubarraySum(int[] nums, int k) {
        Dictionary<int, int> prefixSums = new Dictionary<int, int>();
      
        prefixSums.Add(0, 1);
      
        int count = 0;
        int sum = 0;
      
        foreach (int num in nums) {
            sum += num;
          
            int value = 0;
            if (prefixSums.TryGetValue(sum - k, out value)) {
                count += value;
            }
          
            if (prefixSums.TryGetValue(sum, out value)) {
                prefixSums[sum]++;
            } else {
                prefixSums[sum] = 1;
            }
        }
      
        return count;
    }
}

